### PR TITLE
[scalability] Migrate performance tests to E2

### PR DESCRIPTION
**Pull request by jprzychodzen**
/cc @mborsz 
/assign @wojtek-t 

Migrate performance tests to E2 instance family.

### CODE REVIEW

**Comment by jprzychodzen** [comment]
I've tested manually that master job passes with comparable time

File: config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml [hunk 1]
```diff
@@ -33,6 +33,7 @@ presubmits:
         - --cluster=
         - --extract=local
         - --flush-mem-after-build=true
+        - --gcp-node-size=e2-medium
```
Comment (wojtek-t) [inline config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml:36]: I'm a bit reluctant to changing presubmit immediately with postsubmits. We should split them into two separate PR.

Also @mborsz as current and next-week oncall - FYI

Comment (jprzychodzen) [reply config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml]: Sure, removed presubmit from this PR, I will create another one for presubmits.

Comment (jprzychodzen) [reply config/jobs/kubernetes/sig-scalability/sig-scalability-presubmit-jobs.yaml]: Presubmit only PR is available in #19602

File: config/jobs/kubernetes/sig-release/release-branch-jobs/1.16.yaml [hunk 2]
```diff
@@ -939,6 +940,7 @@ presubmits:
         - --cluster=
         - --extract=local
         - --flush-mem-after-build=true
+        - --gcp-node-size=e2-medium
```
Comment (wojtek-t) [inline config/jobs/kubernetes/sig-release/release-branch-jobs/1.16.yaml:943]: Sorry - missed that. I don't think we should be changing presubmits especially for release branches in the first step...

**Comment by wojtek-t** [comment]
Closing in favor of https://github.com/kubernetes/test-infra/pull/20043

/close
